swarm/storage/mru/resource.go

```diff
@@ -719,78 +756,49 @@ func (h *Handler) update(ctx context.Context, addr storage.Address, data []byte,
 	// an update can be only one chunk long; data length less header and signature data
 	// 12 = length of header and data length fields (2xuint16) plus period and frequency value fields (2xuint32)
 	datalimit := h.chunkSize() - int64(signatureLength-len(rsrc.name)-12)
-	if int64(len(data)) > datalimit {
-		return nil, NewError(ErrDataOverflow, fmt.Sprintf("Data overflow: %d / %d bytes", len(data), datalimit))
-	}
-
-	// get our blockheight at this time and the next block of the update period
-	currentblock, err := h.getBlock(ctx, rsrc.name)
-	if err != nil {
-		return nil, NewError(ErrIO, fmt.Sprintf("Could not get block height: %v", err))
-	}
-	nextperiod, err := getNextPeriod(rsrc.startBlock, currentblock, rsrc.frequency)
-	if err != nil {
-		return nil, err
+	if int64(len(mru.data)) > datalimit {
```

Comment:
Is this correct still, with the new bytes introduced in the updates header?

Reply:
data lengths are taken care during the encoding refactor in #704